### PR TITLE
chore: remove `Promise.allSettled` shim

### PR DESCRIPTION
**Pull request by tenthirtyone**
This was TODO'd to be removed if we bumped `typescript` to `4.2.3+`

### CODE REVIEW

**Review by MicaiahReid — APPROVED** [review]
Looks good to me! I don't think you needed to push the shrinkwrap for this one since no dependencies changed. I don't think it will hurt anything, though.

File: src/packages/core/src/server.ts
Comment (davidmurdoch) [inline src/packages/core/src/server.ts]: The `promise.allsettled` package itself should be removed.

**Review by davidmurdoch — CHANGES_REQUESTED** [review]
The `promise.allsettled` package itself should be removed.

**Comment by davidmurdoch** [comment]
Looks like it doesn't work in Node v12.0.0. Let's wait on this until after the-merge is in

**Comment by davidmurdoch** [comment]
@tenthirtyone let's merge this one _after_ the merge is released (which should be later today).

**Review by jeffsmale90 — COMMENTED** [review]
This all looks good to me. If you merge `develop` it'll kill off the node12 builds 🥳

**Comment by tenthirtyone** [comment]
> This all looks good to me. If you merge `develop` it'll kill off the node12 builds partying_face

@jeffsmale90 - should have been taken care of in https://github.com/trufflesuite/ganache/pull/4003/commits/6268b367cfdb4b3e86b6a37267d3a86b59c824c0

**Review by davidmurdoch — APPROVED** [review]

**Pull request merged**
